src/components/dimensions/DimensionSelect.js

```diff
@@ -1,32 +1,42 @@
 import { DimensionsPanel } from '@dhis2/analytics'
+import { useDataQuery } from '@dhis2/app-runtime'
 import i18n from '@dhis2/d2-i18n'
-import { Popover, IconChevronDown24 } from '@dhis2/ui'
+import { Popover, IconChevronDown24, Help } from '@dhis2/ui'
 import PropTypes from 'prop-types'
-import React, { Fragment, useRef, useState, useEffect } from 'react'
-import { connect } from 'react-redux'
-import { loadDimensions } from '../../actions/dimensions.js'
+import React, { useRef, useState } from 'react'
+import { useUserSettings } from '../UserSettingsProvider.js'
 import styles from './styles/DimensionSelect.module.css'
 
-const DimensionSelect = ({
-    dimension,
-    dimensions,
-    onChange,
-    loadDimensions,
-}) => {
-    const dropdownRef = useRef()
-    const [isOpen, setIsOpen] = useState(false)
+// Include the following dimension types
+const dimensionTypes = [
+    'CATEGORY',
+    'CATEGORY_OPTION_GROUP_SET',
+    'ORGANISATION_UNIT_GROUP_SET',
+]
 
-    useEffect(() => {
-        if (!dimensions) {
-            loadDimensions()
-        }
-    }, [dimensions, loadDimensions])
+// Load above dimension types
+const DIMENSIONS_QUERY = {
+    dimensions: {
+        resource: 'dimensions',
+        params: ({ nameProperty }) => ({
+            fields: ['id', `${nameProperty}~rename(name)`, 'dimensionType'],
+            filter: `dimensionType:in:[${dimensionTypes.join(',')}]`,
+            order: 'displayName:asc',
```

Comment:
also here, will `displayName` always work?

Reply:
Changed to `${nameProperty}:asc`